### PR TITLE
Update to socket2 v0.4

### PR DESCRIPTION
**Pull request by Thomasdezeeuw**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ default-target = "x86_64-pc-windows-msvc"
 targets = ["aarch64-pc-windows-msvc", "i686-pc-windows-msvc", "x86_64-pc-windows-msvc"]
 
 [dependencies]
-socket2 = "0.3.16"
+socket2 = "0.4.0"
```
Comment (faern) [inline Cargo.toml:22]: As far as I can tell, this is only used in the tests. I think we can move this to `[dev-dependencies]`? We then have to remove `extern crate socket2` from `lib.rs`. But I don't understand what that declaration do there at all. This is a Rust 2018 crate, and it's not publicly re-exported either. Am I missing something? :thinking:

**Review by faern — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (Thomasdezeeuw) [reply Cargo.toml:22]: Likely it's due to the age of the code, written pre Rust 2018. I can remove them if you want.

Comment (faern) [reply Cargo.toml:22]: I looked at the open PRs from before I became a maintainer of this crate. There was already a PR on moving `socket2` to a dev-dependency. So I merged that one just now. But it's still not upgraded. So could you please rebase on latest master and still do the upgrade to `0.4`?

Comment (Thomasdezeeuw) [reply Cargo.toml]: Done.

**Comment by faern** [comment]
Awesome! Since it's a dev dependency it won't end up in everyone's dependency tree in the same way. But still great to have the latest. Thanks :tada:

**Comment by faern** [comment]
Published 0.3.7 :rocket:

**Pull request merged**